### PR TITLE
Support count Expression Extensions

### PR DESCRIPTION
**Pull request by jpogran**
This adds count as body schema extensions.

Needs https://github.com/hashicorp/hcl-lang/pull/134 to be merged

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,7 +6,6 @@ require (
 	github.com/google/go-cmp v0.5.9
 	github.com/hashicorp/go-cleanhttp v0.5.2
 	github.com/hashicorp/go-version v1.6.0
-	github.com/hashicorp/hcl-lang v0.0.0-20220823151539-1fce4f11c137
```
Comment (jpogran) [inline go.mod:9]: I'm not sure why this moved when updating to latest hcl-lang

File: internal/schema/0.13/provider_block.go [hunk 2]
```diff
@@ -32,6 +32,9 @@ func providerBlockSchema() *schema.BlockSchema {
 		},
 		Description: lang.PlainText("A provider block is used to specify a provider configuration"),
 		Body: &schema.BodySchema{
+			Extensions: &schema.BodyExtensions{
+				Count: true,
+			},
```
Comment (dbanck) [inline internal/schema/0.13/provider_block.go:37]:
```suggestion
```
I _think_ provider blocks don't support the `count` attribute

File: internal/schema/0.14/variable_block.go [hunk 3]
```diff
@@ -34,6 +34,9 @@ var variableBlockSchema = &schema.BlockSchema{
 	Description: lang.Markdown("Input variable allowing users to customizate aspects of the configuration when used directly " +
 		"(e.g. via CLI, `tfvars` file or via environment variables), or as a module (via `module` arguments)"),
 	Body: &schema.BodySchema{
+		Extensions: &schema.BodyExtensions{
+			Count: true,
+		},
```
Comment (dbanck) [inline internal/schema/0.14/variable_block.go:39]:
```suggestion
```

I _think_ variable blocks don't support the `count` attribute

File: internal/schema/1.2/data.go [hunk 4]
```diff
@@ -9,6 +9,9 @@ func datasourceLifecycleBlock() *schema.BlockSchema {
 	return &schema.BlockSchema{
 		Description: lang.Markdown("Lifecycle customizations to set validity conditions of the datasource"),
 		Body: &schema.BodySchema{
+			Extensions: &schema.BodyExtensions{
+				Count: true,
+			},
```
Comment (dbanck) [inline internal/schema/1.2/data.go:14]:
```suggestion
```
I _think_ lifecycle blocks don't support the `count` attribute

File: internal/schema/1.2/resource.go [hunk 5]
```diff
@@ -11,6 +11,9 @@ func resourceLifecycleBlock() *schema.BlockSchema {
 	return &schema.BlockSchema{
 		Description: lang.Markdown("Lifecycle customizations to change default resource behaviours during plan or apply"),
 		Body: &schema.BodySchema{
+			Extensions: &schema.BodyExtensions{
+				Count: true,
+			},
```
Comment (dbanck) [inline internal/schema/1.2/resource.go:16]:
```suggestion
```
I _think_ lifecycle blocks don't support the `count` attribute

File: internal/schema/0.12/provisioner_block.go [hunk 6]
```diff
@@ -22,7 +22,8 @@ func provisionerBlock(v *version.Version) *schema.BlockSchema {
 			},
 		},
 		Body: &schema.BodySchema{
-			HoverURL: "https://www.terraform.io/docs/language/resources/provisioners/syntax.html",
+			Extensions: &schema.BodyExtensions{},
```
Comment (dbanck) [inline internal/schema/0.12/provisioner_block.go:25]:
```suggestion
```
Do we need an empty `BodyExtensions` block here?

File: internal/schema/0.12/module_block.go [hunk 7]
```diff
@@ -29,6 +29,9 @@ func moduleBlockSchema() *schema.BlockSchema {
 		},
 		Description: lang.PlainText("Module block to call a locally or remotely stored module"),
 		Body: &schema.BodySchema{
+			Extensions: &schema.BodyExtensions{
+				Count: true,
+			},
```
Comment (dbanck) [inline internal/schema/0.12/module_block.go:34]:
```suggestion
```
Count for module blocks was [introduced in 0.13](https://github.com/hashicorp/terraform/blob/v0.13/CHANGELOG.md#0130-august-10-2020). Your change in `internal/schema/0.13/module_block.go` covers this.

File: internal/schema/0.12/connection_block.go [hunk 8]
```diff
@@ -14,6 +14,9 @@ func connectionBlock(v *version.Version) *schema.BlockSchema {
 		MaxItems:               1,
 		SemanticTokenModifiers: lang.SemanticTokenModifiers{tokmod.Connection},
 		Body: &schema.BodySchema{
+			Extensions: &schema.BodyExtensions{
+				Count: true,
+			},
```
Comment (dbanck) [inline internal/schema/0.12/connection_block.go:19]:
```suggestion
```
I _think_ connection blocks don't support the `count` attribute

**Pull request merged**